sw/host/tests/manuf/otp_ctrl/src/main.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+//! This test performs a lifecycle transition from `RAW` to `TEST_UNLOCKED0`.
+
+use anyhow::Context;
+use structopt::StructOpt;
+
+use opentitanlib::app::TransportWrapper;
+use opentitanlib::dif::otp_ctrl::DamParam;
+use opentitanlib::io::jtag::{JtagParams, JtagTap};
+use opentitanlib::test_utils::init::InitializeTest;
+use opentitanlib::test_utils::otp_ctrl::OtpParam;
+
+#[derive(Debug, StructOpt)]
+struct Opts {
+    #[structopt(flatten)]
+    init: InitializeTest,
+    #[structopt(flatten)]
+    jtag: JtagParams,
+}
+
+fn main() -> anyhow::Result<()> {
+    let opts = Opts::from_args();
+    opts.init.init_logging();
+
+    let transport = opts.init.init_target()?;
+    otp_ctrl_test(&opts, &transport)?;
+
+    Ok(())
+}
+
+fn otp_ctrl_test(opts: &Opts, transport: &TransportWrapper) -> anyhow::Result<()> {
+    // Set the TAP straps for the CPU and reset.
+    transport
+        .pin_strapping("PINMUX_TAP_RISCV")?
+        .apply()
+        .context("failed to apply RISCV TAP strapping")?;
+    transport
+        .reset_target(opts.init.bootstrap.options.reset_delay, true)
+        .context("failed to reset")?;
+
+    // Connect to the RISCV TAP via JTAG.
+    let jtag = transport.jtag(&opts.jtag)?;
+    jtag.connect(JtagTap::RiscvTap)
+        .context("failed to connect to RISCV TAP over JTAG")?;
+
+    // Read the `EnCsrngSwAppRead` which should be pre-programmed.
+    let mut en_csrng_sw_app_read = [0x12345678];
+    OtpParam::read_param(
+        &*jtag,
+        DamParam::EnCsrngSwAppRead,
+        &mut en_csrng_sw_app_read,
+    )
+    .context("failed to read parameter EN_CSRNG_SW_APP_READ")?;
+
+    println!("EN_CSRNG_SW_APP_READ: {en_csrng_sw_app_read:#x?}");
+    //assert_eq!(en_csrng_sw_app_read, [0x01]);
+
+    // Program and then read back the `TEST_UNLOCK_TOKEN` which isn't pre-set.
+    let test_unlock_token = [0x89abcdef; 4];
+    OtpParam::write_param(&*jtag, DamParam::TestUnlockToken, &test_unlock_token)
+        .context("failed to write TEST_UNLOCK_TOKEN")?;
```

Comment:
This shouldn't work as you have things configured, i.e., you are using the `//hw/bitstream:rom_otp_dev` bitstream, which uses the [`img_dev` otp image](https://cs.opensource.google/opentitan/opentitan/+/master:hw/ip/otp_ctrl/data/BUILD;drc=dcb3fec2d61a8f05ef4cc5b6fe2938693310ca65;l=458), which has the [STD_OTP_OVERLAYS](https://cs.opensource.google/opentitan/opentitan/+/master:rules/otp.bzl;drc=6d0bcbec6e7cf728436f13c181654b673b43a672;l=346), which includes a [set and locked SECRET0 partition](https://cs.opensource.google/opentitan/opentitan/+/master:hw/ip/otp_ctrl/data/BUILD;drc=dcb3fec2d61a8f05ef4cc5b6fe2938693310ca65;l=231).

If you define an empty OTP image + bitstream to use for this test like I describe above, this should work.